nix/e2e-tests/tests/test_dev_proxy.py

```
@@ -0,0 +1,165 @@
+def _free_port() -> int:
+    s = socket.socket()
+    s.bind(("127.0.0.1", 0))
+    port = s.getsockname()[1]
+    s.close()
+    return port
+
+
+class _CaptureHandler(http.server.BaseHTTPRequestHandler):
+    """Backend that records the request-body length per X-Test-Case."""
+
+    received: dict = {}
```

Ruff (0.15.15)

[warning] Mutable default value for class attribute

(RUF012)